### PR TITLE
Bank Statement Analyzer: View transactions implementation

### PR DESCRIPTION
**Pull request by juliaozm**

### CODE REVIEW

**Comment by juliaozm** [comment]
@pro1code1hack Please review the latest commit

File: data/accounts.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    {
+        "user_id": 3,
+        "account_number": "1122334455",
+        "iban": "FR7630006000011234567890189",
+        "bic": "AGRIFRPP",
+        "sort_code": "30001007",
+        "balance": 750.75
+    }
+]
```
Comment (pro1code1hack) [inline data/accounts.py]: This should be removed

Comment (pro1code1hack) [reply data/accounts.py]: I appreciate that you've genuinely tested this and created mock data, but this should not be a part of the MR

File: data/categories.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    {
+        "name": "Bills",
+        "description": "Monthly payments",
+    },
+    {
+        "name": "Entertainment",
+        "description": "Extra personal expenses",
+    }
+]
```
Comment (pro1code1hack) [inline data/categories.py]: Same here

File: data/statements.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        "account_id": 3,
+        "start_date": "2023-05-01",
+        "end_date": "2023-05-31",
+        "opening_balance": 6000.00,
+        "closing_balance": 7000.00,
+        "money_out": 500.00,
+        "money_in": 1500.00
+    },
+]
```
Comment (pro1code1hack) [inline data/statements.py]: Same here

File: data/users.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        "name": "Alice Johnson",
+        "email": "alice.johnson@example.com",
+        "password": "password123"
+    },
+    {
+        "name": "Jane Tompson",
+        "email": "jane.tompson@example.com",
+        "password": "password123"
+    }
+]
```
Comment (pro1code1hack) [inline data/users.py]: Same here

File: db.py [hunk 5]
```diff
@@ -19,7 +19,22 @@ def connect(self) -> sqlite3.Connection:
...
+            """
+            DROP TABLE IF EXISTS statements;
+            """,
+            """
+            DROP TABLE IF EXISTS categories;
+            """,
+            """
+            DROP TABLE IF EXISTS transactions;
+            """,
+            """
```
Comment (pro1code1hack) [inline db.py]: I will update sql queries to the `CREATE TABLE IF NOT EXISTS`

Comment (pro1code1hack) [reply db.py]: Actually, could you do this for me please, as a part of this MR

Comment (pro1code1hack) [reply db.py]: Jus restore statements back, no need to drop tables as we might have existing data

File: transactions/transaction_service.py [hunk 6]
```diff
@@ -35,7 +35,63 @@ def __init__(self, db: sqlite3.Connection) -> None:
...
+            WHERE t.transaction_date 
+            BETWEEN ? AND ?;
+        """
+        try:
+            cursor = self.db.connection.cursor()
+            cursor.execute(insert_sql, [start_day, end_day])
+            rows = cursor.fetchall()
+            return rows
+        except Error as e:
+            print(e)
```
Comment (pro1code1hack) [inline transactions/transaction_service.py]: raise an error instead of printing

**Review by pro1code1hack — CHANGES_REQUESTED** [review]

File: data/accounts.py (hunk 1 above)
Comment (juliaozm) [reply data/accounts.py]: Updated

File: data/categories.py (hunk 2 above)
Comment (juliaozm) [reply data/categories.py]: Updated

File: data/statements.py (hunk 3 above)
Comment (juliaozm) [reply data/statements.py]: Updated

File: data/users.py (hunk 4 above)
Comment (juliaozm) [reply data/users.py]: Updated

File: db.py (hunk 5 above)
Comment (juliaozm) [reply db.py]: Updated

File: transactions/transaction_service.py (hunk 6 above)
Comment (juliaozm) [reply transactions/transaction_service.py]: Updated

**Comment by juliaozm** [comment]
Please review again @pro1code1hack

**Pull request merged**